agent/api/task/task_linux.go

```diff
@@ -265,6 +265,15 @@ func (task *Task) initializeFSxWindowsFileServerResource(cfg *config.Config, cre
 	return errors.New("task with FSx for Windows File Server volumes is only supported on Windows container instance")
 }
 
+func (task *Task) getTaskUser(container *container.Container) string {
```

Comment:
This code will return empty user string for Linux when the conditions are not met.
That is not how it is in presently for Linux.

Reply:
I got this verified by the Linux team. They don't care about what user it is. I also checked with them on the change of the sequence of operation too. They agreed with the change.

Reply:
No this is incorrect. Overwriting this config with empty user can have unintended impact.
Unless you can provide a strong reason why this should be allowed, this is a breaking change.